File: 06-errors/00-comparison-with-javascript/main.js
function main() {
  try {
    const user = getUser();
    // use the "user" here

    // profile has to be here within the scope of user, it cannot be in it's own trycatch block
    const profile = getUserProfile(user.ID);
    // use the "profile" here
  } catch (err) {
    console.log(err);
    // the issue here is that we cannot handle the errors individually they all end up here
  }
}

// this function signature doesn't tell us what is returned, and also that the function can throw
function getUser() {
  // do some get user logic here
  return user;
}

File: 06-errors/00-comparison-with-javascript/main.go
package main

import "fmt"

func main() {
	user, err := getUser()
	if err != nil {
		fmt.Println(err)
		return
	}
	// use the "user" here

	profile, err := getUserProfile(user.ID)
	// ^this depends on the function above for its argument user.ID
	if err != nil {
		fmt.Println(err)
		return
	}
	// use the "profile" here
}

// by looking at this function signature we can see it returns a User and possibly an error we need to handle
func getUser() (User, err) {
	// do some get user logic here
}

// Go Error handling forces you to be hyper aware of every potential error and that you write code which handles it
